api/src/wfl/service/datarepo.clj

```diff
@@ -97,6 +97,15 @@
   ([job-id]
    (poll-job job-id 5)))
 
+(defn job-done?
```

Comment:
I really really don't like this style of writing predicates. Likewise, I hate it when `throw-if-foo` functions return the foo when it's valid. 
I'd be fine if it was called something else like `get-result-when-done`. I'd be fine with two functions `job-done?` and `get-job-result` to be used in a `when` which might make sense as you'll want to find the first job that's finished in the queue.

Reply:
Is job-done still used somewhere?

Reply:
>Is job-done still used somewhere?

Ah good point, it's not used anymore, so I will probably remove it. (The peek/pop will likely have to implement something similar though)